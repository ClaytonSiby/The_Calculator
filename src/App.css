body {
  padding: 0;
  margin: 0;
  height: 100vh;
  width: 100%;
  overflow: hidden;
  background: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url('./assets/main_bg.jpg');
  background-size: cover;
  background-position: center center;
  background-repeat: no-repeat;
}

#root {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
}

.mainWrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

nav {
  width: 100%;
  display: flex;
  justify-content: space-between;
  background: rgba(0, 0, 0, 0.7);
  font-family: sans-serif;
  padding: 0 1rem;
  color: grey;
}

ul {
  height: 100%;
  padding: 0 0.8rem;
}

li {
  padding: 0.7rem 2rem;
  float: left;
  font-size: 1.2rem;
  font-weight: 700;
  list-style: none;
}

li a {
  color: #fff;
  text-decoration: none;
}

.quote {
  display: flex;
  height: 40%;
  max-width: 60%;
  align-items: center;
  margin: auto;
  padding: .7rem;
  background: rgba(0, 0, 0, .5);
}

.quote p {
  font-size: 1.6rem;
  color: #fff;
}

.homeContainer {
  display: flex;
  height: 40%;
  max-width: 60%;
  align-items: center;
  margin: auto;
  padding: 0 .7rem;
  background: rgba(0, 0, 0, .5);
  color: #fff;
  font-family: sans-serif;
}
