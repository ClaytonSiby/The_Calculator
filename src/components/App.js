import React, { useState } from 'react'
import Display from './Display'
import ButtonPanel from './ButtonPanel'
import calculate from '../logic/calculate'
import styles from '../assets/styles.module.css'
import calcStyles from '../assets/app.module.css'

const App = () => {
  const initialState = { total: null, next: null, operation: null }
  const [state, setState] = useState(initialState)

  const handleClick = btnName => {
    setState(calculate(state, btnName))
  }

  const { total, next, operation } = state

  return (
    <div className='mainWrapper'>
      <div className={`${styles.flexContainer} ${calcStyles.calcContainer}`}>
        <Display total={total} next={next} operation={operation} />
        <ButtonPanel clickHandler={handleClick} />
      </div>
    </div>
  )
}

export default App
