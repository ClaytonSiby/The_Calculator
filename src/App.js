import './App.css';

function App() {
  return (
    <div className="App">
      <h1>Milestone1 Project Setup</h1>
    </div>
  );
}

export default App;
