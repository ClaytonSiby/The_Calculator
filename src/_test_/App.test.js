import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from '../components/App';

describe('App component', () => {
  test('renders learn react link', () => {
    render(<App />);
  });

  test('renders output consistently to the DOM', () => {
    const test = renderer.create(<App />).toJSON();
    expect(test).toMatchSnapshot();
  });
});
