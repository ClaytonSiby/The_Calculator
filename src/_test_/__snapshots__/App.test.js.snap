// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders output consistently to the DOM 1`] = `
<div
  className="mainWrapper"
>
  <div
    className="flexContainer calcContainer"
  >
    <div
      className="flexContainer displayComp"
    >
      <p>
        <span>
          0
        </span>
        <br />
        <span>
           
           
        </span>
      </p>
    </div>
    <div
      className="flexContainer"
    >
      <div
        className="fixedHeight"
      >
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": false,
              "width": "25%",
            }
          }
          type="button"
        >
          AC
        </button>
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": false,
              "width": "25%",
            }
          }
          type="button"
        >
          +/-
        </button>
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": false,
              "width": "25%",
            }
          }
          type="button"
        >
          %
        </button>
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="fixedHeight"
      >
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": false,
              "width": "25%",
            }
          }
          type="button"
        >
          7
        </button>
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": false,
              "width": "25%",
            }
          }
          type="button"
        >
          8
        </button>
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": false,
              "width": "25%",
            }
          }
          type="button"
        >
          9
        </button>
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="fixedHeight"
      >
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": false,
              "width": "25%",
            }
          }
          type="button"
        >
          4
        </button>
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": false,
              "width": "25%",
            }
          }
          type="button"
        >
          5
        </button>
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": false,
              "width": "25%",
            }
          }
          type="button"
        >
          6
        </button>
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="fixedHeight"
      >
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": false,
              "width": "25%",
            }
          }
          type="button"
        >
          1
        </button>
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": false,
              "width": "25%",
            }
          }
          type="button"
        >
          2
        </button>
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": false,
              "width": "25%",
            }
          }
          type="button"
        >
          3
        </button>
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="fixedHeight"
      >
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": false,
              "width": "50%",
            }
          }
          type="button"
        >
          0
        </button>
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": false,
              "width": "25%",
            }
          }
          type="button"
        >
          .
        </button>
        <button
          className="maxWidth"
          onClick={[Function]}
          style={
            Object {
              "background": "orange",
              "width": "25%",
            }
          }
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
